vdev/src/commands/generate/publish_metadata.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::Result;
+use chrono::prelude::*;
+
+use crate::{git, util};
+use std::env;
+use std::fs::OpenOptions;
+use std::io::{self, Write};
+
+/// Setting necessary metadata for our publish workflow in CI
+#[derive(clap::Args, Debug)]
+#[command()]
+pub struct Cli {}
+
+impl Cli {
+    pub fn exec(self) -> Result<()> {
+        // Generate the Vector version and build description.
+        let version = util::read_version()?;
+
+        let git_sha = git::get_git_sha()?;
+        let current_date = Local::today().naive_local().to_string();
+        let build_desc = format!("{git_sha} {current_date}");
+
+        // Figure out what our release channel is.
+        let channel = util::release_channel()?;
+
+        // Depending on the channel, this influences which Cloudsmith repository we publish to.
+        let cloudsmith_repo = match channel {
+            "nightly" => "vector-nightly",
+            _ => "vector"
+        };
+
+        // Set the output variables
+        let version_output = format!("vector_version={version}");
+        let build_desc_output = format!("vector_build_desc={build_desc}");
+        let channel_output = format!("vector_release_channel={channel}");
+        let cloudsmith_repo_output = format!("vector_cloudsmith_repo={cloudsmith_repo}");
```

Comment:
I don't see there being much point to formatting these here and then outputting them using a format string below.

Reply:
You're right I will remove the unnecessary format! call